program/src/instruction.rs

```diff
@@ -880,6 +880,38 @@ pub enum MangoInstruction {
     /// 2. `[signer]` owner_ai - Owner of Mango Account
     /// 3. `[]` delegate_ai - delegate
     SetDelegate,
+
+    /// Make a flash loan.
+    ///
+    /// Accounts expected: 4 + any optional
+    ///
+    ///   . `[]` mango_group_ai - MangoGroup
+    ///   . `[]` signer_ai - Source token account authority
+    ///   . `[writable]` source_token_ai - Source token account from which loan would be acquired
+    ///   . `[writable]` destination_token_ai - Destination token account
+    ///   . `[]` token_prog_ai - Token program id.
+    ///   . `[]` flash_loan_prog_ai - Flash loan receiver program id.
```

Comment:
I think receiving program having to implement instruction with tag 0 is too restrictive. Why not pass in the ix data?